WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewViewModel.swift

```diff
@@ -30,28 +47,51 @@ final class ProductDetailPreviewViewModel: ObservableObject {
     @Published var isShowingViewPhotoSheet = false
     @Published var notice: Notice?
 
-    /// Temporary logic check to enable the undo option for product name
+    var optionsTitle: String {
+        guard let generatedAIProduct else {
+            return ""
+        }
+
+        return String.localizedStringWithFormat(Localization.OptionSwitch.title,
+                                                selectedOptionIndex + 1,
+                                                generatedAIProduct.names.count)
+    }
+
+    var canSwitchBetweenOptions: Bool {
+        options.count > 1
+    }
+
+    var canSelectPreviousOption: Bool {
+        selectedOptionIndex > 0
+    }
+
+    var canSelectNextOption: Bool {
+        guard let generatedAIProduct else {
+            return false
+        }
+
+        return selectedOptionIndex < generatedAIProduct.names.count - 1
+    }
+
     var hasChangesToProductName: Bool {
-        guard let generatedProduct else {
+        guard let generatedAIProduct else {
```

Comment:
Should avoid crashes by ensuring we can subscript the names array as well?

```suggestion
        guard let generatedAIProduct, generatedAIProduct.names.count > selectedOptionIndex  else {
```

Similar comment for the other fields.

Reply:
I started reading values from the array using `safe:` in ebc6bcd862f40711308fb3c76045f1757384d620